lib/src/fnc/math.rs

```diff
@@ -137,7 +143,17 @@ pub fn spread(_: &Context, mut args: Vec<Value>) -> Result<Value, Error> {
 }
 
 pub fn sqrt(_: &Context, mut args: Vec<Value>) -> Result<Value, Error> {
-	Ok(args.remove(0).as_number().sqrt().into())
+	let v = args.remove(0);
+	let v_number = v.as_number();
+	let v_float = v_number.clone().as_float();
+
+	match v_float {
+		num if num >= 0.0 => Ok(v_number.sqrt().into()),
+		_ => Err(Error::InvalidArguments {
+			name: String::from("math::sqrt"),
+			message: String::from("The argument must be a positive number."),
+		}),
+	}
```

Comment:
@tsunyoku when running in schemaless mode, if a field doesn't exist, currently this would cause an error, but should probably just return `Value::None`.

Could we change this to...

```rs
pub fn sqrt(_: &Context, mut args: Vec<Value>) -> Result<Value, Error> {
	match args.remove(0).as_number() {
 		v if v >= Number::Int(0) => Ok(v.sqrt().into()),
		_ => Ok(Value::None),
 	}
}
```

Reply:
ah, oops. will change